refocus-bdk-client.js

```diff
@@ -720,6 +721,44 @@ module.exports = (config) => {
         roomObject, TOKEN);
     }, // updateRoomName
 
+    getOrInitializeBotData: (room, botName, dataName, defaultValue) => {
+      log.debug('Getting or Initialize BotData. ', { room, botName, dataName });
+      return new Promise((resolve, reject) => {
+        return genericGet(`${SERVER}${API}${BOTDATA_ROUTE}` +
+        `?roomId=${room}&botId=${botName}&name=${dataName}`, TOKEN)
+          .then((data) => {
+            const _data = data.body.filter((bd) =>
+              bd.name === dataName)[FIRST_ARRAY_EL];
+            return _data;
+          })
+          .then((data) => {
+            if (data) {
+              const botDataValue = data.value;
+              return botDataValue;
+            }
+            let newData = defaultValue;
+            if (newData && typeof newData !== 'string') {
+              newData = serialize(newData);
+            }
+            const botData = {
+              'name': dataName,
+              'roomId': parseInt(room, 10),
+              'botId': botName,
+              'value': newData
+            };
+            return genericPost(`${SERVER}${API}${BOTDATA_ROUTE}`,
+              botData, TOKEN)
+              .then(() => newData);
```

Comment:
What happens if there is something wrong with this post? Should the client always rely on 'newData'?

Reply:
Maybe we should also be returning the value from the response from refocus, not just what the user gave as an argument

Reply:
This is the same pattern all of the bots have been doing for their data when its initialised.